File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/Microsoft ADO.NET Data Services Application/VB/UserInterface/Gateways/IProductGateway.vb
' Copyright © Microsoft Corporation.  All Rights Reserved.
' This code released under the terms of the 
' Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports AdventureWorksService


Namespace UserInterface.Gateways
	Public Interface IProductGateway
		Function GetProducts(ByVal productName As String, ByVal category As ProductCategory) As IList(Of Product)
		Function GetCategories() As IList(Of ProductCategory)
		Function DeleteProduct(ByVal product As Product) As String
		Sub UpdateProduct(ByVal product As Product)
		Sub AddProduct(ByVal product As Product)
	End Interface
End Namespace

File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/Microsoft ADO.NET Data Services Application/C#/UserInterface/Gateways/IProductGateway.cs
// Copyright © Microsoft Corporation.  All Rights Reserved.
// This code released under the terms of the 
// Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using UserInterface.AdventureWorksService;


namespace UserInterface.Gateways
{
    public interface IProductGateway
    {
        IList<Product> GetProducts(string productName, ProductCategory category);
        IList<ProductCategory> GetCategories();
        string DeleteProduct(Product product);
        void UpdateProduct(Product product);
        void AddProduct(Product product);
    }
}
